purchases/src/main/kotlin/com/revenuecat/purchases/common/FileHelper.kt

```diff
@@ -25,29 +32,36 @@ internal class FileHelper(
         return file.delete()
     }
 
-    fun readFilePerLines(filePath: String): List<String> {
-        val readLines = mutableListOf<String>()
-        val file = getFileInFilesDir(filePath)
-        FileInputStream(file).use { fileInputStream ->
-            InputStreamReader(fileInputStream).use { inputStreamReader ->
-                BufferedReader(inputStreamReader).use { bufferedReader ->
-                    readLines.addAll(bufferedReader.readLines())
-                }
+    fun readFilePerLines(filePath: String, dataListener: DataListener<Pair<String, Int>>) {
```

Comment:
I created this listener to be able to simulate a "Stream" of data with a completion.

BufferedReader provides the `lines` method: https://developer.android.com/reference/java/io/BufferedReader#lines() which would have been very useful, but it's API 24+ only... 

We could make diagnostics be API 24+ only, but I thought it would be good to gather data from older devices as well, so decided to just implement our own thing.

Reply:
Unless I was looking at it wrong, it seems like API 24+ has like 98% usage though?

Reply:
It would be 95.4%... It's pretty high, but not sure if we want to introduce that limitation on diagnostics... I think it's not that hard to support older clients so I would leave it as is. Lmk if you feel strongly about it though

<img width="523" alt="image" src="https://github.com/RevenueCat/purchases-android/assets/808417/c1d1866d-0775-4b7b-b500-1436d4b3d692">

Reply:
The thing is, that's 5% of _current_ users, possibly using old app versions. How many of those are still getting updates?
For comparison, I'm making diagnostics on iOS iOS 13.0+ only which came out in 2019. Level 24 is 2016, right?

Just saying that if it makes code easier, I doubt we'd even notice the difference by bumping the diagnostics API requirement.